### PR TITLE
Use InMemory channel in functional tests

### PR DESCRIPTION
**Pull request by lmolkova**
~~supposedly~~ fixes issues #1153, #1154, #1127, #1096.

func tests use servertelemetrychannel, but it does not guarantee to immediately send telemetry. 
Let's try using InMemory channel and see if it stabilizes tests.

### CODE REVIEW

**Comment by lmolkova** [comment]
All these changes:
- InMemoryChannel in dependency and web func tests
- Forcing docker image recreation 

seem to help a lot reducing the failure rate to ~25%.
still there is some problem that sometimes test app container is not started properly/timely/etc. I'll create another issue for that.

**Comment by MS-TimothyMothra** [comment]
This LGTM, but because @cijothomas is the author, I want to give him a chance to review. :)

File: Test/Web/FunctionalTests/TestApps/AspNetDiagnostics/ApplicationInsights.config [hunk 1]
```diff
@@ -42,9 +42,9 @@
     <Add Type="Microsoft.ApplicationInsights.WindowsServer.UnobservedExceptionTelemetryModule, Microsoft.AI.WindowsServer"/>
     <Add Type="Microsoft.ApplicationInsights.WindowsServer.FirstChanceExceptionStatisticsTelemetryModule, Microsoft.AI.WindowsServer"/>
   </TelemetryModules>
-  <TelemetryChannel Type="Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel.ServerTelemetryChannel, Microsoft.AI.ServerTelemetryChannel">
+  <TelemetryChannel Type="Microsoft.ApplicationInsights.Channel.InMemoryChannel, Microsoft.ApplicationInsights">
     <EndpointAddress>http://localhost:4000/v2/track</EndpointAddress>
-    <MaxTelemetryBufferDelay>0:00:01</MaxTelemetryBufferDelay>
+    <DeveloperMode>true</DeveloperMode>
```
Comment (cijothomas) [inline Test/Web/FunctionalTests/TestApps/AspNetDiagnostics/ApplicationInsights.config]: if devmode is true, then we dont need flush at all right?

File: Test/E2ETests/E2ETests/Test452Base.cs [hunk 2]
```diff
@@ -74,12 +74,13 @@ public static void MyClassInitializeBase()
...
 
-            //DockerUtils.RemoveDockerImage(Apps[AppNameBeingTested].imageName, true);
-            //DockerUtils.RemoveDockerContainer(Apps[AppNameBeingTested].containerName, true);
+            DockerUtils.RemoveDockerImage(Apps[AppNameBeingTested].imageName, true);
+            DockerUtils.RemoveDockerContainer(Apps[AppNameBeingTested].containerName, true);
 
             // Deploy the docker cluster using Docker-Compose
-            //DockerUtils.ExecuteDockerComposeCommand("up -d --force-recreate --build", DockerComposeFileName);
-            DockerUtils.ExecuteDockerComposeCommand("up -d --build", DockerComposeFileName);
+            DockerUtils.ExecuteDockerComposeCommand("up -d --force-recreate --build", DockerComposeFileName);
```
Comment (cijothomas) [inline Test/E2ETests/E2ETests/Test452Base.cs]: if we are force-recreating, the the above steps to remove image and container is not required.
but if this is helping anyway - lets just do it.

File: Test/Web/FunctionalTests/TestApps/AspNetDiagnostics/ApplicationInsights.config (hunk 1 above)
Comment (lmolkova) [reply Test/Web/FunctionalTests/TestApps/AspNetDiagnostics/ApplicationInsights.config]: this is Web, there is no flush there. But I guess in dependency tracking functional tests yes, we don't. need it.

In theory, it should with ServerTelemteryChannel isn't it?

Comment (cijothomas) [reply Test/Web/FunctionalTests/TestApps/AspNetDiagnostics/ApplicationInsights.config]: Yes devmode=true means we should be flushing instantly after every item automatically.

**Pull request merged**